### PR TITLE
Add git clone only branch

### PR DESCRIPTION
**Pull request by Angelk90**
# Description

# Checklist

- [x] I read and understood the [contributing guide](https://github.com/AuroraEditor/AuroraEditor/blob/main/CONTRIBUTING.md) as well as the [code of conduct](https://github.com/AuroraEditor/AuroraEditor/blob/main/CODE_OF_CONDUCT.md)
- [x] My changes generate no new warnings
- [x] My code builds and runs on my machine
- [x] I documented my code
- [x] Review requested

# Screenshots

https://user-images.githubusercontent.com/20476002/189291452-80214560-3191-4b73-93a2-a2e5e94ab3d1.mov

### CODE REVIEW

**Comment by nanashili** [comment]
Don't put remote code in the branches file, i created a remote file specifically for remotes

**Comment by wdg** [comment]
is this still a WIP?

File: AuroraEditor/Base/Git/UI/Clone/GitCloneView.swift [hunk 1]
```diff
@@ -69,28 +124,40 @@ public struct GitCloneView: View {
...
+                        check = 0
+                        activeSheet = .verify
+                    }.sheet(item: $activeSheet) { item in
+                        switch item {
+                        case .verify:
+                            progressVerifyView
+                        case .select:
+                            selectView
+                        case .error:
+                            errorView
```
Comment (wdg) [inline AuroraEditor/Base/Git/UI/Clone/GitCloneView.swift:157]:
```suggestion
```

(remove this)

File: AuroraEditor/Base/Git/UI/Clone/GitCloneView.swift [hunk 2]
```diff
@@ -43,6 +59,45 @@ public struct GitCloneView: View {
...
+        } catch {
+            Log.error("Failed to find main branch name.")
+        }
+    }
+
+    func getRemoteBranch(url: String) {
+        do {
+            let branches = try Remote().getRemoteBranch(url: url)
+            if branches[0].contains("fatal:") {
+                Log.info("Error: getRemoteBranch")
```
Comment (wdg) [inline AuroraEditor/Base/Git/UI/Clone/GitCloneView.swift:86]:
```suggestion
                Log.error("Error: getRemoteBranch")
```

**Comment by nanashili** [comment]
This PR is starting to take to long with no actuve work really being done on it. Can you get this done @Angelk90 or we need to close it and open it at a later date.

File: AuroraEditor/Base/Git/UI/Clone/GitCloneView.swift [hunk 3]
```diff
@@ -106,9 +174,102 @@ public struct GitCloneView: View {
...
+                    context[.leading]
+                }
+            }
+        }
+        .padding(.top, 20)
+        .padding(.horizontal, 20)
+        .padding(.bottom, 16)
+        .frame(width: 400, height: 150)
+        .onAppear {
+            DispatchQueue.main.asyncAfter(deadline: .now() + 3) {
```
Comment (wdg) [inline AuroraEditor/Base/Git/UI/Clone/GitCloneView.swift:212]:
```suggestion
            DispatchQueue.global(qos: .background).async {
```

File: AuroraEditor/Base/Git/UI/Clone/GitCloneView.swift [hunk 4]
```diff
@@ -41,21 +56,53 @@ public struct GitCloneView: View {
...
+            if branches[0].contains("fatal:") {
+                Log.warning("Error: getRemoteBranch")
+                activeSheet = .error("Error: getRemoteBranch")
+            } else {
+                self.arrayBranch = branches
+                self.check += 1
+                if check == 2 {
+                    check = 0
+                    activeSheet = .select
+                }
```
Comment (wdg) [inline AuroraEditor/Base/Git/UI/Clone/GitCloneView.swift:94]:
```suggestion
                if check == 2 {
                    DispatchQueue.main.asyncAfter(deadline: .now() + 3) {
                      check = 0
                      activeSheet = .select
                    }
                }
```

**Review by wdg — CHANGES_REQUESTED** [review]

**Review by nanashili — APPROVED** [review]

**Review by wdg — APPROVED** [review]

**Pull request merged**